src/lib/kfs_access.c

```diff
@@ -1,7 +1,10 @@
 #include <kfs_access.h>
 
-int kfs_access(const char* path, int mask) {
-    int grant = 1;
-    if(grant) return(0);
-    return(-EACCES);
+int kfs_access(const char* original_path, int mode) {
+    char *path = local_disk_cache_path(original_path);
+    int res = access(path, mode);
+    if (res == -1) {
+        return -errno;
+    }
+    return 0;
```

Comment:
why not return res?

Reply:
function dont access file its rather define if next function have permission to perform tested function